src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleEqualityTests.cs

```diff
@@ -1878,12 +1932,18 @@ void M(S s1, S s2)
 }";
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics(
-                // (6,35): error CS0306: The type 'S' may not be used as a type argument
-                //         System.Console.Write(("", s1) == (null, s2));
-                Diagnostic(ErrorCode.ERR_BadTypeArgument, "s1").WithArguments("S").WithLocation(6, 35),
-                // (6,30): error CS0019: Operator '==' cannot be applied to operands of type 'S' and 'S'
-                //         System.Console.Write(("", s1) == (null, s2));
-                Diagnostic(ErrorCode.ERR_BadBinaryOps, @"("""", s1) == (null, s2)").WithArguments("==", "S", "S").WithLocation(6, 30)
+                    // (6,35): error CS0306: The type 'S' may not be used as a type argument
+                    //         System.Console.Write(("", s1) == (null, s2));
+                    Diagnostic(ErrorCode.ERR_BadTypeArgument, "s1").WithArguments("S").WithLocation(6, 35),
+                    // (6,30): error CS0019: Operator '==' cannot be applied to operands of type 'S' and 'S'
+                    //         System.Console.Write(("", s1) == (null, s2));
+                    Diagnostic(ErrorCode.ERR_BadBinaryOps, @"("""", s1) == (null, s2)").WithArguments("==", "S", "S").WithLocation(6, 30),
+                    // (6,35): error CS0306: The type 'S' may not be used as a type argument
+                    //         System.Console.Write(("", s1) == (null, s2));
+                    Diagnostic(ErrorCode.ERR_BadTypeArgument, "s1").WithArguments("S").WithLocation(6, 35),
```

Comment:
This appears to be a duplicate warning.

Reply:
[https://github.com/dotnet/roslyn/issues/37121](https://github.com/dotnet/roslyn/issues/37121)

---
In reply to: [301725757](https://github.com/dotnet/roslyn/pull/37052#discussion_r301725757) [](ancestors = 301725757)